readers/api/transport.go

```diff
@@ -82,7 +82,7 @@ func decodeList(ctx context.Context, r *http.Request) (interface{}, error) {
 	}
 
 	if err := authorize(ctx, r, chanID); err != nil {
-		return nil, err
+		return nil, errors.ErrAuthorization
```

Comment:
You are ignoring the custom errors returned by this method and they probably already wrapped with `errors.Authorization`. Can you check it please?